### PR TITLE
cpp backend milestone1 - java frontend upgrade

### PR DESCRIPTION
**Pull request by lxning**
## Description

Please read our [CONTRIBUTING.md](https://github.com/pytorch/serve/blob/master/CONTRIBUTING.md) prior to creating your first pull request.

Please include a summary of the feature or issue being fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #(issue)
#1651 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

Please describe the Unit or Integration tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

add Unit test WorkerLifeCycleTest in CI build.

- [ ] Test A
Logs for Test A

- [ ] Test B
Logs for Test B

## Checklist:

- [ ] Did you have fun?
- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

**Comment by msaroufim** [comment]
FYI you can commit code with `git commit -m "[skip ci]"` and it'll skip the github action from running to save CI cycles

File: frontend/server/src/test/resources/snapshots/snapshot1.cfg [hunk 1]
```diff
@@ -9,7 +9,7 @@ default_workers_per_model=4
 model_store=../archive/src/test/resources/models
 async_logging=true
 number_of_gpu=0
-model_snapshot={\n  "name"\: "20200329045334828-startup.cfg",\n  "modelCount"\: 1,\n  "created"\: 1585457614832,\n  "models"\: {\n    "noop"\: {\n      "1.11"\: {\n        "defaultVersion"\: true,\n        "marName"\: "noop.mar",\n        "minWorkers"\: 4,\n        "maxWorkers"\: 4,\n        "batchSize"\: 1,\n        "maxBatchDelay"\: 100,\n        "responseTimeout"\: 120\n      }\n    }\n  }\n}
+model_snapshot={\n  "name"\: "20200329045334828-startup.cfg",\n  "modelCount"\: 1,\n  "created"\: 1585457614832,\n  "models"\: {\n    "noop"\: {\n      "1.11"\: {\n        "defaultVersion"\: true,\n        "marName"\: "noop.mar",\n        "minWorkers"\: 4,\n        "maxWorkers"\: 4,\n        "batchSize"\: 1,\n        "maxBatchDelay"\: 100,\n        "responseTimeout"\: 120,\n        "runtimeType"\: "python"\n     }\n    }\n  }\n}
```
Comment (mreso) [inline frontend/server/src/test/resources/snapshots/snapshot1.cfg:12]: What happens if we do not have RuntimeType="python" in the cfg file? Is this covered with a test as well?

**Review by mreso — COMMENTED** [review]
Hi Li, I left some comments, let me know if you have any questions.

**Comment by msaroufim** [comment]
PR seems stale so closing for now

**Pull request closed without merging**